### ISSUE TITLE
XMLHttpRequest timeout issue

### ISSUE THREAD

**Issue by albertdaurell** [rollbar/rollbar.js#948]
Hi, I was recently trying to throttle the browser network and I found that rollbar is not handling timeouts because `xmlhttprequest.timeout` is not set, so this [xdomainrequest.ontimeout](https://github.com/rollbar/rollbar.js/blob/master/src/browser/transport.js#L185) or another timeout  callback is never _called_.

```javascript
xdomainrequest.ontimeout = function() { // Never thrown
  var msg = 'Request timed out';
  var code = 'ETIMEDOUT';
  callback(_newRetriableError(msg, code));
};
```

From [XMLHttpRequest.timeout](https://developer.mozilla.org/en/docs/Web/API/XMLHttpRequest/timeout) we can read: _The default value is 0, which means there is no timeout_

 From some embedded devices this can be a problem because it seems the implementation from the device itself can have issues reporting if the network had some temporary issues, requests gets "stalled" and it never sends the data to rollbar.

### Proposal

Adding some extra parameter in the main configuration:

```javascript
var rollbarConfig = {
      maxRetries: 3,
      retryInterval: 10000,
      timeout: 5000,
      ...
    };
```

And setting this `timeout` after [request.open(method, url, true);](https://github.com/rollbar/rollbar.js/blob/master/src/browser/transport.js#L159)

I left an example there:

https://github.com/rollbar/rollbar.js/pull/947

#### Without the changes

1. Open the browser
2. Load rollbar
3. Throttle the network to 1kb/s
4. Call `Rollbar.log`
5. Wait for 10 seconds
5. Data is not sent

#### With my proposal

1. Open the browser
2. Load rollbar with config.timeout = 5000 and config.retryInterval = 10000
3. Throttle the network to 1kb/s
4. Call `Rollbar.log`
5. See that timeout is reached and retry after 10 seconds is "executed"

**Comment by albertdaurell** [comment]
Any update on this?

**Comment by waltjones** [comment]
@albertdaurell Yes it would be good to support this. I'll take a closer look at the PR today.

**Comment by albertdaurell** [comment]
@waltjones  Also I'm thinking about adding maxRetries parameter, something like this:

<img src="https://user-images.githubusercontent.com/7487579/119527313-7fa64a00-bd80-11eb-8c65-034c8a175199.png" width="400px" />

**Comment by albertdaurell** [comment]
Thx @waltjones added maxRetries also to the pull

**Comment by albertdaurell** [comment]
@waltjones Should the issue also be closed? 

BTW: Thank you very much for your support and patience!

**Comment by waltjones** [comment]
Yes, this will be released in v2.22.0 tomorrow. Nice work, and thank you!
